resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/utils/FileUtil.java

```diff
@@ -340,6 +347,71 @@ public static final void transferStreams(InputStream source, OutputStream destin
 		}
 	}
 
+	public static char[] readAllChars(IFile file) throws CoreException {
+		byte[] content = file.readAllBytes();
+		Charset charset = getCharset(file, content);
+		return FileUtil.toCharArray(content, charset);
+	}
+
+	public static String readString(IFile file) throws CoreException {
+		byte[] content = file.readAllBytes();
```

Comment:
this can use `readAllChars()`

Reply:
Though it could, I'd rather avoid converting the bytes to chars here such that we can benefit from compressed strings (represented as bytes internally) without the intermediate allocation of the char[]

Reply:
That would introduce ephemeral memory allocation for the char array which is not needed otherwise. This is a hotspot and optimized for speed.